modules/oe_list_pages_open_vocabularies/src/SearchApiConfigurator.php

```diff
@@ -0,0 +1,157 @@
+use Drupal\oe_list_pages\ListSourceFactoryInterface;
+use Drupal\open_vocabularies\OpenVocabularyAssociationInterface;
+use Drupal\search_api\Entity\Index;
+use Drupal\search_api\Item\Field;
+use Drupal\search_api\Item\FieldInterface;
+
+/**
+ * Configure search api when vocabulary associations are created.
+ */
+class SearchApiConfigurator {
+
+  /**
+   * The entity type manager.
+   *
+   * @var Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The list source factory.
+   *
+   * @var Drupal\oe_list_pages\ListSourceFactoryInterface
+   */
+  protected $listSourceFactory;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, ListSourceFactoryInterface $listSourceFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->listSourceFactory = $listSourceFactory;
+  }
+
+  /**
+   * Updates the search api configuration.
+   *
+   * It adds a search index field and facet for an open vocabulary association.
+   *
+   * @param \Drupal\open_vocabularies\OpenVocabularyAssociationInterface $association
+   *   The openvocabulary association.
+   * @param string $field_id
+   *   The field_id.
+   */
+  public function updateConfig(OpenVocabularyAssociationInterface $association, string $field_id): void {
+    $field_config_storage = $this->entityTypeManager->getStorage('field_config');
+    // Get the correct index looking to the list source.
+    $field_config = $field_config_storage->load($field_id);
+    $entity_type = $field_config->getTargetEntityTypeId();
+    $bundle = $field_config->getTargetBundle();
+    /** @var \Drupal\oe_list_pages\ListSourceInterface $list_source */
+    $list_source = $this->listSourceFactory->get($entity_type, $bundle);
+    /** @var \Drupal\search_api\IndexInterface $index */
+    $index = $list_source->getIndex();
+
+    // Generates the field and facet id.
+    $id = 'open_vocabularies_' . str_replace('.', '_', $association->id() . '_' . $field_id);
+
+    // Creates the field.
+    /** @var \Drupal\search_api\Field $index_field */
+    $index_field = $this->getField($index, $id);
+    $index_field->setType('string');
+    $index_field->setPropertyPath($field_config->getFieldStorageDefinition()->getName());
+    $index_field->setDatasourceId('entity:' . $entity_type);
+    $index_field->setLabel($association->label());
+    $index->save();
+
+    // Creates the facet.
+    /** @var \Drupal\facets\FacetInterface $facet */
+    $facet = $this->getFacet($id);
+    $facet->setUrlAlias($id);
+    $facet->set('name', $association->label());
+    $facet->addProcessor([
+      'processor_id' => 'url_processor_handler',
+      'weights' => [],
+      'settings' => [],
+    ]);
+    $facet->setEmptyBehavior([]);
+    $facet->setFacetSourceId($list_source->getSearchId());
+    $facet->setWidget('oe_list_pages_multiselect', []);
+    $facet->setFieldIdentifier($id);
+    $facet->save();
```

Comment:
* You are missing the facet weight. It's a required field in the form (can be 0)
* Don't we need the `Transform entity ID to label` processor as well?
* The `reference` facet also has the `Hide facet when facet source is not rendered` processor enabled 
* We should also configure some default sorting behaviour.